src/bvals/neighbor_block.hpp

```diff
@@ -0,0 +1,182 @@
+//========================================================================================
+// (C) (or copyright) 2020-2021. Triad National Security, LLC. All rights reserved.
+//
+// This program was produced under U.S. Government contract 89233218CNA000001 for Los
+// Alamos National Laboratory (LANL), which is operated by Triad National Security, LLC
+// for the U.S. Department of Energy/National Nuclear Security Administration. All rights
+// in the program are reserved by Triad National Security, LLC, and the U.S. Department
+// of Energy/National Nuclear Security Administration. The Government is granted for
+// itself and others acting on its behalf a nonexclusive, paid-up, irrevocable worldwide
+// license in this material to reproduce, prepare derivative works, distribute copies to
+// the public, perform publicly and display publicly, and to permit others to do so.
+//========================================================================================
+#ifndef BVALS_NEIGHBOR_BLOCK_HPP_
+#define BVALS_NEIGHBOR_BLOCK_HPP_
+//! \file neighbor_block.hpp
+//  \brief defines enums, structs, and abstract classes
+
+// TODO(felker): deduplicate forward declarations
+// TODO(felker): consider moving enums and structs in a new file? bvals_structs.hpp?
+
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "parthenon_mpi.hpp"
+
+#include "defs.hpp"
+#include "parthenon_arrays.hpp"
+#include "utils/error_checking.hpp"
+
+namespace parthenon {
+
+// forward declarations
+class Mesh;
+class MeshBlock;
+class MeshBlockTree;
+class Field;
+class ParameterInput;
+struct RegionSize;
+
+// TODO(felker): nest these enum definitions inside bvals/ classes, when possible.
+
+// DEPRECATED(felker): maintain old-style (ALL_CAPS) enumerators as unscoped,unnamed types
+// Keep for compatibility with user-provided pgen/ files. Use only new types internally.
+
+// GCC 6 added Enumerator Attr (v6.1 released on 2016-04-27)
+// TODO(felker): replace with C++14 [[deprecated]] attributes if we ever bump --std=c++14
+#if (defined(__GNUC__) && __GNUC__ >= 6) || (defined(__clang__) && __clang_major__ >= 3)
+enum {
+  FACE_UNDEF __attribute__((deprecated)) = -1,
+  INNER_X1 __attribute__((deprecated)),
+  OUTER_X1 __attribute__((deprecated)),
+  INNER_X2 __attribute__((deprecated)),
+  OUTER_X2 __attribute__((deprecated)),
+  INNER_X3 __attribute__((deprecated)),
+  OUTER_X3 __attribute__((deprecated))
+};
+enum {
+  BLOCK_BNDRY __attribute__((deprecated)) = -1,
+  BNDRY_UNDEF __attribute__((deprecated)),
+  REFLECTING_BNDRY __attribute__((deprecated)),
+  OUTFLOW_BNDRY __attribute__((deprecated)),
+  PERIODIC_BNDRY __attribute__((deprecated))
+};
+#else
+enum { FACE_UNDEF = -1, INNER_X1, OUTER_X1, INNER_X2, OUTER_X2, INNER_X3, OUTER_X3 };
+enum {
+  BLOCK_BNDRY = -1,
+  BNDRY_UNDEF,
+  REFLECTING_BNDRY,
+  OUTFLOW_BNDRY,
+  USER_BNDRY,
+  PERIODIC_BNDRY,
+  POLAR_BNDRY,
+  POLAR_BNDRY_WEDGE
+};
+#endif
```

Comment:
I thought about this but decided against doing anything because I really wasn't sure how important this backwards compatibility was to downstream codes.

Reply:
I *think* it's completely unused. Though I could be wrong about that. @pgrete maybe can chime in.

Reply:
Just checked and we don't use it downstream, so this can go.